src/App.tsx

```diff
@@ -1,25 +1,44 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
-
 // eslint-disable-next-line @typescript-eslint/no-unused-vars
+
 const items = getNumbers(1, 42).map(n => `Item ${n}`);
 
 export const App: React.FC = () => {
+  const [split, setSplit] = useState(5);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const total = items.length;
+
+  const pageFrom = split * (currentPage - 1);
+  const pageTo = split * currentPage;
+  const visibleItems = [...items].slice(pageFrom, pageTo);
```

Comment:
```suggestion
  const visibleItems = items.slice(pageFrom, pageTo);
```
slice doesn't mutate the array, so it's redundant to make a copy